api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java

```diff
@@ -0,0 +1,1127 @@
+    private static final int DEMO_LOCATION1_ID = 1;
+    private static final int DEMO_LOCATION2_ID = 2;
+    
+    public EncounterSearchAdviserTest() {
+    }
+    
+    @Override
+    protected void setInterceptorAndServices(TestWithAOP testCase) {
+        testCase.setInterceptor(UserServiceInterceptorAdvice.class);
+        testCase.addService(UserService.class);
+    }
+    
+    @Before
+    public void setUp()  throws Exception {
+        encounterService = Context.getEncounterService();
+        userService = Context.getUserService();
+        locationService = Context.getLocationService();
+        patientService = Context.getPatientService();
+        visitService = Context.getVisitService();
+        executeDataSet(XML_FILENAME_WITH_ADMIN_USER_DATA);
+        executeDataSet(XML_FILENAME_WITH_PERSON_DATA);
+    }
+    
+    @Test
+    public void getEncounter_adminCanAccessEncounterFromAllLocations() {
+        Context.authenticate("admin", "test");
+        User authenticatedUser = Context.getAuthenticatedUser();
+        assertNotNull(authenticatedUser);
+        assertTrue(authenticatedUser.isSuperUser());
+        
+        Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+        Location location2 = locationService.getLocation(DEMO_LOCATION2_ID);
+                
+        Encounter encounter1 = encounterService.getEncounter(1);
+        assertNotNull(encounter1);
+        assertEquals(encounter1.getLocation(),location1);
+        
+        Encounter encounter2 = encounterService.getEncounter(2);
+        assertNotNull(encounter2);
+        assertEquals(encounter2.getLocation(),location2);
+        
+        Encounter encounter3 = encounterService.getEncounter(3);
+        assertNotNull(encounter3);
+        assertEquals(encounter3.getLocation(),location1);
+        
+        Encounter encounter4 = encounterService.getEncounter(4);
+        assertNotNull(encounter4);
+        assertEquals(encounter4.getLocation(),location1);
+        
+        Encounter encounter5 = encounterService.getEncounter(5);
+        assertNotNull(encounter5);
+        assertEquals(encounter5.getLocation(),location2);
+        
+        Encounter encounter6 = encounterService.getEncounter(6);
+        assertNotNull(encounter6);
+        assertEquals(encounter6.getLocation(),location2);
+        
+        Encounter encounter7 = encounterService.getEncounter(7);
+        assertNotNull(encounter7);
+        assertEquals(encounter7.getLocation(),location2);
+        
+        Encounter encounter8 = encounterService.getEncounter(8);
+        assertNotNull(encounter8);
+        assertEquals(encounter8.getLocation(),location2);
+    }
+    
+    @Test
+    public void getEncounter_nonAdminUserWithOutLocationPropertyShouldNotGetEncounterInformation(){
+        User normalUser = userService.getUser(DEMO_USER1_ID);
+        Context.authenticate(normalUser.getUsername(), "userServiceTest");
+        assertFalse(normalUser.isSuperUser());

```

Comment:
Here your test case indicated that it should be a normal user. Then why are you authenticating as an Admin (super user) here?